code/modules/antagonists/mind_flayer/powers/flayer_stealth_powers.dm

```diff
@@ -0,0 +1,245 @@
+
+/datum/spell/flayer/skin_suit/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.include_user = FALSE
+	T.allowed_type = /mob/living
+	T.try_auto_target = TRUE
+	T.click_radius = -1
+	T.selection_type = SPELL_SELECTION_VIEW
+	return T
+
+/datum/spell/flayer/skin_suit/cast(list/targets, mob/living/user)
+	var/mob/living/target = targets[1]
+	user.apply_status_effect(STATUS_EFFECT_MAGIC_DISGUISE, target)
+
+/datum/spell/flayer/skin_suit/on_apply()
+	..()
+	cooldown_handler.recharge_duration -= 30 SECONDS
+
+/// After a 7 second channel time you can emag a borg
+/datum/spell/flayer/self/override_key
+	name = "Silicon Administrative Access"
+	desc = "Allows us to charge our hand with a mass of nanites that hijacks cyborgs lawsets."
+	action_icon_state = "magnet" // Uhhhhhhhhhhhhhhhhhhhhhhhhhhh
+	power_type = FLAYER_PURCHASABLE_POWER
+	category = FLAYER_CATEGORY_INTRUDER
+	base_cooldown = 2 SECONDS //The cast time is going to be the main limiting factor, not cooldown
+	base_cost = 150
+	stage = 3
+	var/hand_type = /obj/item/melee/swarm_hand
+
+/datum/spell/flayer/self/override_key/cast(list/targets, mob/user)
+	if(istype(user.l_hand, hand_type))
+		qdel(user.l_hand)
+		flayer.send_swarm_message("We dissipate the nanites.")
+		return FALSE
+	if(istype(user.r_hand, hand_type))
+		qdel(user.r_hand)
+		flayer.send_swarm_message("We dissipate the nanites.")
+		return FALSE
+
+	var/obj/item/melee/swarm_hand/funny_hand = new hand_type
+	if(!user.put_in_hands(funny_hand))
+		flayer.send_swarm_message("Our hands are currently full.")
+		qdel(funny_hand)
+		return FALSE
+
+/obj/item/melee/swarm_hand
+	name = "Nanite Mass"
+	desc = "Will attempt to convert any cyborg you touch into a loyal member of the hive after a 7 second delay."
+	icon = 'icons/obj/weapons/magical_weapons.dmi'
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	color = COLOR_BLACK
+	flags = ABSTRACT | DROPDEL
+	w_class = WEIGHT_CLASS_HUGE
+	var/conversion_time = 7 SECONDS
+
+/obj/item/melee/swarm_hand/afterattack(atom/target, mob/living/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!isrobot(target))
+		return
+	var/mob/living/silicon/robot/borg = target
+	target.visible_message(
+		"<span class='danger'>[user] puts [user.p_their()] hands on [target] and begins transferring energy!</span>",
+		"<span class='userdanger'>[user] puts [user.p_their()] hands on [target] and begins transferring energy!</span>")
+	if(borg.emagged || !borg.is_emaggable)
+		to_chat(user, "<span class='notice'>Your override attempt fails before it can even begin.</span>")
+		qdel(src)
+		return
+	if(!do_mob(user, borg, conversion_time))
+		to_chat(user, "<span class='notice'>Your concentration breaks.</span>")
+		qdel(src)
+		return
+	to_chat(user, "<span class='notice'>The mass of swarms vanish into the cyborg's internals. Success.</span>")
+	qdel(src)
```

Comment:
I think you might need to set `src = null` or somehow detach? This feels a little sus

Reply:
This should do something like `INVOKE_ASYNC` really, I'll do it later